Nagstamon/QUI/__init__.py

```diff
@@ -1210,6 +1210,16 @@ def show_window(self, event=None):
 
                     self.show()
 
+                    # Using the EWMH protocol to move the window to the active desktop.
+                    if not platform.system() in ('Darwin', 'Windows'):
+                        winid = self.winId().__int__()
+                        deskid = self.ewmh.getCurrentDesktop()
+                        self.ewmh.setWmDesktop(winid, deskid)
+                        self.ewmh.display.flush()
+
+                    # makes the window manager switch to the desktop where this widget has appeared
+                    self.raise_()
```

Comment:
Is the desktop really changing here? I intended to open the popup at the desktop where the user clicks the systray icon.